modules/tinymce/src/core/main/ts/Rtc.ts

```diff
@@ -0,0 +1,296 @@
+import { Event, Node as DomNode, Range } from '@ephox/dom-globals';
+import { Fun, Obj, Option, Type } from '@ephox/katamari';
+import Editor from './api/Editor';
+import Node from './api/html/Node';
+import Serializer from './api/html/Serializer';
+import * as FilterNode from './html/FilterNode';
+import * as Operations from './undo/Operations';
+import { Index, Locks, UndoBookmark, UndoLevel, UndoLevelType, UndoManager } from './undo/UndoManagerTypes';
+import * as ApplyFormat from './fmt/ApplyFormat';
+import * as RemoveFormat from './fmt/RemoveFormat';
+import * as ToggleFormat from './fmt/ToggleFormat';
+import { RangeLikeObject } from './selection/RangeTypes';
+import { FormatRegistry } from './fmt/FormatRegistry';
+import { Content } from './content/EditorContent';
+import { GetContentArgs, ContentFormat, getContentInternal } from './content/GetContentImpl';
+import { SetContentArgs, setContentInternal } from './content/SetContentImpl';
+import { insertHtmlAtCaret } from './content/InsertContentImpl';
+import { getSelectedContentInternal } from './selection/GetSelectionContentImpl';
+
+const isTreeNode = (content: any): content is Node => content instanceof Node;
+
+const isSupportedContentFormat = (format: string) => format !== 'text';
+
+/** API implemented by the RTC plugin */
+interface RtcRuntimeApi {
+  undo: () => void;
+  redo: () => void;
+  hasUndo: () => boolean;
+  hasRedo: () => boolean;
+  transact: (fn: () => void) => void;
+  applyFormat: (format: string, vars: Record<string, string>) => void;
+  removeFormat: (format: string, vars: Record<string, string>) => void;
+  toggleFormat: (format: string, vars: Record<string, string>) => void;
+  getContent: () => Node | null;
+  setContent: (node: Node) => void;
+  insertContent: (node: Node) => void;
+  getSelectedContent: () => Node | null;
+  getRawModel: () => any;
+  isRemote: boolean;
+}
+
+/** A copy of the TinyMCE api definitions that the plugin overrides  */
+interface RtcAdaptor {
+  undoManager: {
+    beforeChange: (locks: Locks, beforeBookmark: UndoBookmark) => void
+    addUndoLevel: (undoManager: UndoManager, index: Index, locks: Locks, beforeBookmark: UndoBookmark, level?: UndoLevel, event?: Event) => UndoLevel
+    undo: (undoManager: UndoManager, locks: Locks, index: Index) => UndoLevel
+    redo: (index: Index, data: UndoLevel[]) => UndoLevel
+    clear: (undoManager: UndoManager, index: Index) => void
+    reset: (undoManager: UndoManager) => void
+    hasUndo: (undoManager: UndoManager, index: Index) => boolean
+    hasRedo: (undoManager: UndoManager, index: Index) => boolean
+    transact: (undoManager: UndoManager, locks: Locks, callback: () => void) => UndoLevel
+    ignore: (locks: Locks, callback: () => void) => void
+    extra: (undoManager: UndoManager, index: Index, callback1: () => void, callback2: () => void) => void
+  };
+  formatter: {
+    applyFormat: (format: string, vars?: Record<string, string>, node?: DomNode | RangeLikeObject) => void;
```

Comment:
Is there a reason these were renamed from the current API names? eg `remove` vs `removeFormat`? As it'd be less confusing if it was the same name to make it clear that's what it's implementing/delegating for.

Reply:
Also a case of restructuring (this used to be flat, rather than nested objects). Thanks for the pickup!

- [ ] rename back